### PR TITLE
refactor: update `blas/ext/base/scusum` to follow current project conventions

### LINKED ISSUE #1516: [RFC]: refactor `blas/ext/base/scusum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/scusum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/scusum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AuenKr**
Resolves #1516 .

## Description

> What is the purpose of this pull request?

This pull request:

- refactors `@stdlib/blas/ext/base/scusum`  to follow current projects convention

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1516 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by AuenKr** [comment]
@Pranavchiku, can you review the PR.

**Review by kgryte — CHANGES_REQUESTED** [review]
I'll go ahead and mark this PR as needing changes, as discussed in https://github.com/stdlib-js/stdlib/pull/1867. Whatever changes are made there will need to be made in this PR, as well.

**Comment by AuenKr** [comment]
@kgryte, applied all suggestions.

**Comment by AuenKr** [comment]
@kgryte, waiting for review.

**Comment by kgryte** [comment]
@AuenKr Thank you for the ping. Sorry, I've been a bit swapped. I will try and review this within the next day.

File: lib/node_modules/@stdlib/blas/ext/base/scusum/manifest.json
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/scusum/manifest.json]:
```suggestion
        "stdlib/napi/argv-float",
```

The initial value should be extracted as a single-precision number.

File: lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c]:
```suggestion
#include "stdlib/napi/argv_float.h"
```

File: lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_FLOAT( env, N, argv, 1 );
```

File: lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c]:
```suggestion
	status = napi_create_double( env, stdlib_strided_scusum( N, sum, X, strideX, Y, strideY ), &v );
```

You don't need to perform explicit casts (see, e.g., https://github.com/stdlib-js/stdlib/blob/2c3c709353a0821a187ff63251a296f558f951a1/lib/node_modules/%40stdlib/blas/base/scopy/src/addon.c#L41). The `NAPI_ARGV_STRIDED_*` macros handle that for you.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks, @AuenKr, for working on this. I left a couple of minor comments. I'll leave it to you to resolve, as the comments may apply to other PRs which you may have open making similar changes to strided array interfaces.

File: lib/node_modules/@stdlib/blas/ext/base/scusum/manifest.json
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/scusum/manifest.json]:
```suggestion
  "options": {
    "task": "build"
  },
```

You need to specify the default "task".

**Comment by AuenKr** [comment]
@kgryte, applied all changes to other PR also.

File: lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c]: Wait...sorry. I missed this in `dcusum`. What is the return value of `stdlib_strided_scusum`? I believe we aren't returning any value. And in the `addon.cpp`, we returned a `nullptr` (see https://github.com/stdlib-js/stdlib/pull/1864/files#diff-9e113d4e7147f5a08bf0703c64a82b428dcfa4b9468802df73ad1d4e3cc1d73aL152).

As such, this implementation is incorrect.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c]: For an example of what I'd expect, see https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/dfill/src/addon.c.

You can also open up a patch for `dcusum` with the requested changes.

Comment (AuenKr) [reply lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c]: I will soon fix this and open the patch also.

**Comment by AuenKr** [comment]
@kgryte, applied changes.

File: lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c
Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c]: Great! Thanks!

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @AuenKr!

**Pull request merged**
